src/place-screen/components/QuaySection.tsx

```diff
@@ -69,14 +69,16 @@ export function QuaySection({
     mode === 'Favourite'
       ? departures.sort((a, b) => compareByLineNameAndDesc(t, a, b))
       : departures;
+
+  const navigateToQuayEnabled = !!navigateToQuay;
+
+  // If the user has toggled "Show only favorites" and there are no favorites on
+  // this quay, we should minimize the quay section.
+  const hasFavorites = !!favoriteDepartures.find((f) => quay.id === f.quayId);
   useEffect(() => {
     if (!showOnlyFavorites) return setIsMinimized(false);
-    setIsMinimized(
-      !!navigateToQuay &&
-        !favoriteDepartures.find((favorite) => quay.id === favorite.quayId),
-    );
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [showOnlyFavorites]);
+    setIsMinimized(navigateToQuayEnabled && !hasFavorites);
+  }, [showOnlyFavorites, navigateToQuayEnabled, hasFavorites]);
```

Comment:
Nice to simplify the hook by dragging out calculations outside of it. Could even this be possible?

```ts
  const shouldBeMinimized =
    showOnlyFavorites && navigateToQuayEnabled && !hasFavorites;
  useEffect(() => {
    setIsMinimized(shouldBeMinimized);
  }, [shouldBeMinimized]);
```

Reply:
That seems to work like a charm 🙌